component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/EmailOtpServiceImpl.java

```diff
@@ -548,4 +571,213 @@ private int getTenantId() {
+
+        if (NumberUtils.isNumber(claimValues.get(Constants.EMAIL_OTP_FAILED_ATTEMPTS_CLAIM))) {
+            return Integer.parseInt(claimValues.get(Constants.EMAIL_OTP_FAILED_ATTEMPTS_CLAIM));
+        }
+        return 0;
+    }
+
+    private int getFailedLoginLockoutCount(Map<String, String> claimValues) {
+
+        if (NumberUtils.isNumber(claimValues.get(Constants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM))) {
+            return Integer.parseInt(claimValues.get(Constants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM));
+        }
+        return 0;
+    }
+
+    private void populateAccountLockClaims(long unlockTimePropertyValue, int failedLoginLockoutCountValue,
+                                           Map<String, String> updatedClaims) {
+
+        // Calculate unlock time by adding current time and unlock time interval in milliseconds.
+        long unlockTime = System.currentTimeMillis() + unlockTimePropertyValue;
+        updatedClaims.put(Constants.ACCOUNT_LOCKED_CLAIM, Boolean.TRUE.toString());
+        updatedClaims.put(Constants.EMAIL_OTP_FAILED_ATTEMPTS_CLAIM, "0");
+        updatedClaims.put(Constants.ACCOUNT_UNLOCK_TIME_CLAIM, String.valueOf(unlockTime));
+        updatedClaims.put(Constants.FAILED_LOGIN_LOCKOUT_COUNT_CLAIM,
+                String.valueOf(failedLoginLockoutCountValue + 1));
+        updatedClaims.put(Constants.ACCOUNT_LOCKED_REASON_CLAIM_URI, Constants.MAX_EMAIL_OTP_ATTEMPTS_EXCEEDED);
+        IdentityUtil.threadLocalProperties.get().put(Constants.ADMIN_INITIATED, false);
+    }
+
+    private Map<String, String> getUserClaimValues(User user, String[] claims) throws EmailOtpServerException {
+
+        try {
+            AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) EmailOtpServiceDataHolder
+                    .getInstance().getRealmService().getTenantUserRealm(getTenantId()).getUserStoreManager();
+            return userStoreManager.getUserClaimValues(user.getDomainQualifiedUsername(), claims,
+                    UserCoreConstants.DEFAULT_PROFILE);
+        } catch (UserStoreException e) {
+            log.error("Error while reading user claims.", e);
+            throw Utils.handleServerException(Constants.ErrorMessage.SERVER_USER_STORE_MANAGER_ERROR,
+                    String.format("Failed to read user claims for user ID : %s.", user.getUserID()), e);
+        }
+    }
+
+    private void setUserClaimValues(User user, Map<String, String> updatedClaims) throws EmailOtpServerException {
+
+        try {
+            AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) EmailOtpServiceDataHolder
+                    .getInstance().getRealmService().getTenantUserRealm(getTenantId()).getUserStoreManager();
+            userStoreManager.setUserClaimValues(user.getDomainQualifiedUsername(), updatedClaims,
+                    UserCoreConstants.DEFAULT_PROFILE);
+        } catch (UserStoreException e) {
+            log.error("Error while updating user claims", e);
```

Comment:
Why is the error logged in here specifically?

Reply:
removed